modules/sigprofiler/1.0/sigprofiler.smk

```diff
@@ -0,0 +1,207 @@
+        context_type = '96,DINUC,ID',
+        exome = lambda w: {'genome': 'False', 'capture': 'True'}[w.seq_type],
+        min_sig = 1,
+        max_sig = lambda w: {'SBS96': 20, 'DBS78': 15, 'ID83': 10}[w.type],
+        nmf_repl = 30,
+        norm = 'gmm',
+        nmf_init = 'nndsvd_min',
+        outpath = CFG["dirs"]["estimate"]+"{seq_type}--{genome_build}/{file}"
+    conda: CFG["conda_envs"]["sigprofiler"]
+    threads: CFG["threads"]["estimate"]
+    shell:
+        op.as_one_line("""
+        python {input.script} {input.mat} {params.outpath} 
+        {params.ref} {params.context_type} {params.exome} {params.min_sig} {params.max_sig} 
+        {params.nmf_repl} {params.norm} {params.nmf_init} {threads}
+        """)
+
+rule _sigprofiler_run_extract:
+    input:
+        unpack(get_dir),
+        stat = str(rules._sigprofiler_run_estimate.output.stat)
+    output:
+        decomp = CFG["dirs"]["extract"]+"{seq_type}--{genome_build}/{file}/{type}/Suggested_Solution/COSMIC_{type}_Decomposed_Solution/De_Novo_map_to_COSMIC_{type}.csv"
+    params:
+        ref = lambda w: {"grch37":"GRCh37", "hg19":"GRCh37", 
+                         "grch38": "GRCh38", "hg38": "GRCh38"}[w.genome_build],
+        context_type = '96,DINUC,ID',
+        exome = lambda w: {'genome': 'False', 'capture': 'True'}[w.seq_type],
+        nmf_repl = 200,
+        norm = 'gmm',
+        nmf_init = 'nndsvd_min',
+        outpath = CFG["dirs"]["extract"]+"{seq_type}--{genome_build}/{file}"
+    conda: CFG["conda_envs"]["sigprofiler"]
+    threads: CFG["threads"]["extract"]
+    shell:
+        op.as_one_line("""
+        python {input.script} {input.mat} {params.outpath} 
+        {params.ref} {params.context_type} {params.exome}
+        $(awk 'BEGIN {{OFS=FS=","}} $1 ~ "*" {{S=substr($1,1,length($1)-1); if (S-2<1) {{print 1}} else {{print S-2}}}}' {input.stat})
+        $(awk 'BEGIN {{OFS=FS=","}} $1 ~ "*" {{S=substr($1,1,length($1)-1); print S+2}}' {input.stat})
+        {params.nmf_repl} {params.norm} {params.nmf_init} {threads}
+        """)
+
+# Symlinks the final output files into the module results directory (under '99-outputs/')
+rule _sigprofiler_output_tsv:
+    input:
+        decomp = str(rules._sigprofiler_run_extract.output.decomp)
+    output:
+        decomp = CFG["dirs"]["outputs"] + "cosmic_sigs/{seq_type}--{genome_build}/{file}.COSMIC_{type}.De_Novo_map_to_COSMIC_{type}.csv"
+    run:
+        op.relative_symlink(input.decomp, output.decomp)
+
+# Generates the target sentinels for each run, which generate the symlinks
+rule _sigprofiler_all:
+    input:
+        expand(
+            expand(
+                CFG["dirs"]["outputs"] + "cosmic_sigs/{seq_type}--{genome_build}/{file}.COSMIC_{{type}}.De_Novo_map_to_COSMIC_{{type}}.csv",
+            zip,
+            seq_type = CFG["mafs"]["seq_type"],
+            genome_build = CFG["mafs"]["projected_build"],
+            file = CFG["mafs"]["filename"]),
+         type = ['SBS96', 'ID83', 'DBS78'])
```

Comment:
type should be set in the config or something, don't you think?

Reply:
I've specified type as a config parameter. I've also added the ability to specify other available k-mers.